frontend/src/app/[locale]/subscribe/actions.ts

```diff
@@ -0,0 +1,137 @@
+
+import { getTranslations } from "next-intl/server";
+import { redirect } from "next/navigation";
+
+import { ValidationErrors } from "src/components/subscribe/SubscriptionForm";
+
+type sendyResponse = {
+  validationErrors: ValidationErrors;
+  errorMessage: string;
+};
+
+export async function subscribeEmail(
+  prevState: sendyResponse,
+  formData: FormData,
+) {
+  const t = await getTranslations("Subscribe");
+
+  const { errorMessage, validationErrors } = await subscribeEmailAction(
+    t,
+    formData,
+  );
+  if (errorMessage ?? validationErrors) {
+    return {
+      errorMessage,
+      validationErrors,
+    };
+  }
+  // Navigate to the sub confirmation page if no error returns short circuit the function
+  redirect(`/subscribe/confirmation`);
+}
+
+export async function subscribeEmailAction(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  t: any,
+  formData: FormData,
+): Promise<sendyResponse> {
+  const schema = z.object({
+    name: z.string().min(1, {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+      message: t("errors.missing_name"),
+    }),
+    email: z
+      .string()
+      .min(1, {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+        message: t("errors.missing_email"),
+      })
+      .email({
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+        message: t("errors.invalid_email"),
+      }),
+  });
+
+  const validatedFields = schema.safeParse({
+    name: formData.get("name"),
+    email: formData.get("email"),
+  });
+
+  // Return early if the form data is invalid (server side validation!)
+  if (!validatedFields.success) {
+    return {
+      errorMessage: "",
+      validationErrors: validatedFields.error.flatten().fieldErrors,
+    };
+  }
+
+  // hp = honeypot, if this field is filled in, the form is likely spam
+  // https://sendy.co/api
+  const rawFormData = {
+    name: formData.get("name") as string,
+    LastName: formData.get("LastName") as string,
+    email: formData.get("email") as string,
+    hp: formData.get("hp") as string,
+  };
+
+  // TODO: Implement the email subscription logic here, putting old SENDY code here for reference
+  // Note: Noone is sure where the api url/key/list ID are at the moment and I believe the intention is
+  // to move away from SENDY to a different service.
+  try {
+    const sendyApiUrl = environment.SENDY_API_URL;
+    const sendyApiKey = environment.SENDY_API_KEY;
+    const list = environment.SENDY_LIST_ID;
+    const requestData = {
+      list,
+      subform: "yes",
+      boolean: "true",
+      api_key: sendyApiKey,
+      hp: rawFormData.hp,
+      name: rawFormData.name,
+      LastName: rawFormData.LastName,
+      email: rawFormData.email,
+    };
+
+    const sendyResponse = await fetch(`${sendyApiUrl}/subscribe`, {
```

Comment:
Seems overkill for me to create a provider class for an API that is called once. If we end up with more than one newsletter form then it would make more sense to me.

Reply:
I would tend to agree there. However, it seems like the system we currently have is kind of designed to be built for one off requests. Do we want to revisit that pattern?

Reply:
Let's discuss! I added to the agenda for Thurs.